src/Analyzers/Microsoft.AspNetCore.Analyzer.Testing/src/TestSource.cs

```diff
@@ -26,9 +26,15 @@ public static TestSource Read(string rawSource)
             for (var i = 0; i < lines.Length; i++)
             {
                 var line = lines[i];
-                var markerStartIndex = line.IndexOf(MarkerStart, StringComparison.Ordinal);
-                if (markerStartIndex != -1)
+
+                while (true)
```

Comment:
Why do we use the while-loop/break pattern here now?

Reply:
A single line can have more than one diagnostics in it. The latter ones were previously being ignored.